### PR TITLE
GDEV-201 add aws-backup module

### PR DESCRIPTION
**Pull request by syousry6**

### CODE REVIEW

File: aws/aws-backup/aws_backup.tf [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+resource "aws_kms_alias" "default" {
+  count         = var.create ? 1 : 0
+  name          = "alias/${replace(local.stage_prefix, var.delimiter, "/")}/aws-backup"
+  target_key_id = join("", aws_kms_key.default.*.id)
+}
+
+# AWS Backup vault
+resource "aws_backup_vault" "vault" {
+  name        = join("-", [local.module_prefix, "vault"])
```
Comment (TysonBeck) [inline aws/aws-backup/aws_backup.tf]: - [x] Add count attribute

File: aws/aws-backup/aws_backup.tf [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  kms_key_arn = join("", aws_kms_key.default.*.arn)
+}
+
+
+# AWS Backup plan
+resource "aws_backup_plan" "plan" {
+  name = join("-", [local.module_prefix, "plan"])
```
Comment (TysonBeck) [inline aws/aws-backup/aws_backup.tf]: - [x] Add count attribute

File: aws/aws-backup/aws_backup.tf [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+      // default = 3 years * 365 days = 1095 days
+      delete_after = var.monthly_delete_after
+    }
+  }
+}
+
+
+# AWS Backup selection - resource arn
+resource "aws_backup_selection" "arn_resource_selection" {
+  iam_role_arn = aws_iam_role.aws_backup_role.arn
```
Comment (TysonBeck) [inline aws/aws-backup/aws_backup.tf]: - [x] Add count attribute

File: aws/aws-backup/aws_backup.tf [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+
+# ----------------------------------------------------------------------------------------------------------------------
+# MODULES / RESOURCES
+# ----------------------------------------------------------------------------------------------------------------------
+
+# KMS key for fsx
+
+resource "aws_kms_key" "default" {
+  count                    = var.create ? 1 : 0
```
Comment (TysonBeck) [inline aws/aws-backup/aws_backup.tf]: - [x] Make count dynamic so kms key can be passed in

File: aws/aws-backup/aws_backup.tf [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+  name        = join("-", [local.module_prefix, "vault"])
+  kms_key_arn = join("", aws_kms_key.default.*.arn)
+}
+
+
+# AWS Backup plan
+resource "aws_backup_plan" "plan" {
+  name = join("-", [local.module_prefix, "plan"])
+
+  rule {
```
Comment (TysonBeck) [inline aws/aws-backup/aws_backup.tf]: - [x] Create dynamic rule with map instead of predefined rules

File: aws/aws-backup/aws_backup.tf [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+}
+
+variable "backup_resource_ids" {
+  type    = list
+  default = null
+}
+
+
+# Backup rules
+variable "daily_cron" {
```
Comment (TysonBeck) [inline aws/aws-backup/aws_backup.tf]: - [x] Remove extra variables when modified for dynamic rule

File: aws/aws-backup/aws_backup.tf [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+}
+
+
+
+variable "selection_tags" {
+  type    = list(any)
+  default = []
+}
+
+
```
Comment (TysonBeck) [inline aws/aws-backup/aws_backup.tf]: - [x] Add variable for passing in KMS key arn

**Review by jeremyodle — CHANGES_REQUESTED** [review]
- [x] Add `aws-backup` to .github Labeler
- [x] Update README

File: aws/aws-backup/aws_backup.tf [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+  policy                   = var.policy
+  tags                     = local.tags
+  description              = "KMS for aws backup"
+  key_usage                = var.key_usage
+  customer_master_key_spec = var.customer_master_key_spec
+}
+
+resource "aws_kms_alias" "default" {
+  count         = var.create ? 1 : 0
+  name          = "alias/${replace(local.stage_prefix, var.delimiter, "/")}/aws-backup"
```
Comment (jeremyodle) [inline aws/aws-backup/aws_backup.tf]: - [x] Use module name instead of hardcoding

**Comment by jeremyodle** [comment]
- [x] Remove Terraform Formatting GitHub commenter from Terraform pipeline

**Review by jeremyodle — CHANGES_REQUESTED** [review]
- [x] Rename module to `backup`

**Review by jeremyodle — APPROVED** [review]

**Review by jeremyodle — DISMISSED** [review]

**Review by jeremyodle — APPROVED** [review]

**Pull request merged**